scripts/deploy/32_eth_autocall_vault.ts

```diff
@@ -0,0 +1,177 @@
+import { run } from "hardhat";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import {
+  CHAINID,
+  WETH_ADDRESS,
+  USDC_ADDRESS,
+  OTOKEN_FACTORY,
+  GAMMA_CONTROLLER,
+  MARGIN_POOL,
+} from "../../constants/constants";
+import { AUCTION_DURATION, PREMIUM_DISCOUNT } from "../utils/constants";
+
+const main = async ({
+  network,
+  deployments,
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment) => {
+  const { BigNumber } = ethers;
+  const { parseEther } = ethers.utils;
+  const { deploy } = deployments;
+  const { deployer, owner, keeper, admin, feeRecipient } =
+    await getNamedAccounts();
+  console.log(`32 - Deploying ETH Autocall Vault on ${network.name}`);
+
+  const chainId = network.config.chainId;
+  if (chainId !== CHAINID.ETH_MAINNET) {
+    console.log(`Error: chainId ${chainId} not supported`);
+    return;
+  }
+
+  const pricer = await deployments.get("OptionsPremiumPricerETHPut");
+
+  console.log(`AutocallVaultWETH pricer @ ${pricer.address}`);
+
+  const strikeSelection = await deploy("ManualStrikeSelectionAutocall", {
+    contract: "ManualStrikeSelection",
+    from: deployer,
+    args: [],
+  });
+
+  console.log(`AutocallVaultWETH strikeSelection @ ${strikeSelection.address}`);
+
+  try {
+    await run("verify:verify", {
+      address: strikeSelection.address,
+      constructorArguments: [],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  const lifecycleTreasury = await deploy("VaultLifecycleTreasury", {
+    contract: "VaultLifecycleTreasury",
+    from: deployer,
+  });
+
+  console.log(`VaultLifeCycleTreasury @ ${lifecycleTreasury.address}`);
+
+  // deploy logic
+  const logicDeployment = await deploy("RibbonAutocallVault", {
+    contract: "RibbonAutocallVault",
+    from: deployer,
+    args: [
+      USDC_ADDRESS[chainId],
+      OTOKEN_FACTORY[chainId],
+      GAMMA_CONTROLLER[chainId],
+      MARGIN_POOL[chainId],
+    ],
+    libraries: {
+      VaultLifecycleTreasury: lifecycleTreasury.address,
+    },
+  });
+
+  console.log(`AutocallVault Logic @ ${logicDeployment.address}`);
+
+  try {
+    await run("verify:verify", {
+      address: lifecycleTreasury.address,
+      constructorArguments: [],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  try {
+    await run("verify:verify", {
+      address: logicDeployment.address,
+      constructorArguments: [
+        USDC_ADDRESS[chainId],
+        OTOKEN_FACTORY[chainId],
+        GAMMA_CONTROLLER[chainId],
+        MARGIN_POOL[chainId],
+      ],
+    });
+  } catch (error) {
+    console.log(error);
+  }
+
+  // Autocall vault specific initArgs
```

Comment:
lets move these to constants.ts

Reply:
https://github.com/ribbon-finance/ribbon-v2/pull/313/commits/62a15070dad5583bac0d30ac5f96e41c55532941